__tests__/nametag.test.tsx

```diff
@@ -2,32 +2,52 @@ import '@testing-library/jest-dom'
 import { render, screen, fireEvent } from '@testing-library/react'
 import { userEvent } from '@testing-library/user-event'
 
-import Page from '../app/main/page'
+import NameTag from '../app/main/NameTag'
+
+const currentNameTag = ['', '', '', ''];
+const nameTagStatus = false;
+const setCurrentNameTag = jest.fn();
+const setNameTagStatus = jest.fn();
+const selectedWaveHand = null;
+const waveHands = [];
+
 
 jest.mock('next/navigation', () => jest.requireActual('next-router-mock'))
 jest.mock('../lib/zoomapi', () => jest.requireActual('../lib/fakezoomapi'));
 
 describe('Page', () => {
   it('renders the heading and input fields', () => {
-    render(<Page />)
-
-    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
+    render(
+      <NameTag
+        currentNameTag={currentNameTag}
+        nameTagStatus={nameTagStatus}
+        setCurrentNameTag={setCurrentNameTag}
+        setNameTagStatus={setNameTagStatus}
+        selectedWaveHand={selectedWaveHand}
+        waveHands={waveHands}
+      />
+    );
     expect(screen.getByText('Full Name')).toBeInTheDocument();
     expect(screen.getByText('Preferred Name')).toBeInTheDocument();
     expect(screen.getAllByText('Select Pronouns')[0]).toBeInTheDocument();
     expect(screen.getByText('Self Disclosure')).toBeInTheDocument();
   })
 
   it('toggles the nametag display on switch change', async () => {
-    render(<Page />)
-
-    let switchInput = screen.getByTestId('colored-switch');
+    render(
+      <NameTag
+        currentNameTag={currentNameTag}
+        nameTagStatus={nameTagStatus}
+        setCurrentNameTag={setCurrentNameTag}
+        setNameTagStatus={setNameTagStatus}
+        selectedWaveHand={selectedWaveHand}
+        waveHands={waveHands}
+      />
+    );
+
+    let switchInput = screen.getByRole('switch')
     expect(switchInput).not.toBeChecked();
-
```

Comment:
According to https://mui.com/material-ui/react-switch/#accessibility, switches/checkboxes should always have associated labels.  Can we make the test check that as well?